ecosystem/node-checker/README.md

```diff
@@ -4,42 +4,135 @@ The Aptos Node Health Checker (NHC) is the reference implementation of a node he
+## tl;dr
+While we highly recommend you read this whole README, you can get NHC working in a basic form by doing the following. This baseline configuration is for a devnet fullnode.
 
-First, check out the repo and navigate here.
+Get a baseline configuration:
+```
+cd /tmp/nhc && wget https://raw.githubusercontent.com/aptos-labs/aptos-core/main/ecosystem/node-checker/configurations/devnet_fullnode.yaml
+```
+
+Run NHC:
+```
+docker run -v /tmp/nhc:/nhc -t aptoslabs/node-checker:nightly /usr/local/bin/aptos-node-checker server run --baseline-node-config-paths /nhc/devnet_fullnode.yaml
+```
+
+Hit it with a request:
+```
+curl 'http://localhost:20121/check_node?node_url=http://mynode.mysite.com&baseline_configuration_name=devnet_fullnode'
+```
+
+## How NHC works
+Before running NHC, it is important to know at a high level how NHC works. In short, NHC runs as a service that you send http requests to in order to run a set of validations against your node. A single NHC instance can be configured to test multiple different node configurations, for example:
+
+- Validator Node running in single node testnet.
+- Public FullNode connected to devnet.
+- Validator Node connected to testnet, e.g. as part of an Aptos Incentivized Testnet.
+
+In all cases, validations are performed compared to a baseline. For example, for the second configuration above, the baseline node might be a node run by the Aptos team that demonstrates optimal performance / participation characteristics. The configuration describes where to find this node (URL + port), what evaluators (e.g. metrics checks, TPS tests, API validations, etc.) NHC should runn, what parameters to use for those evaluators, what name the configuration has, and so on. Your node will be compared to this baseline node.
+
+When you send requests to NHC, you include which configuration you want to validate your node against. This means a request to NHC might look like this:
+```
+curl 'http://nhc.aptoslabs.com/check_node?node_url=http://myfullnode.mysite.com&baseline_configuration_name=devnet_fullnode'
+```
+
+## Getting configurations ready
+In order to run NHC, you must have baseline configurations that it can use. You have two options here:
+
+### Start from a pre-existing configuration
+In [./configuration_examples](./configuration_examples) you can find configurations that work for each of the use cases above and more.
+
+You might want to setup configurations in your host system like this:
+```
+mkdir /etc/nhc
+cd /etc/nhc
+configs=(single_node_validator devnet_fullnode ait2_validator); for c in ${configs[@]}; do wget https://raw.githubusercontent.com/aptos-labs/aptos-core/main/ecosystem/node-checker/configurations/$c.yaml; done
+```
+
+These configurations are not quite ready to be used as they are, you will need to modify certain fields, such as the node address or evaluator set used. The best way to iterate on this is to just try to run NHC with the configuration and see what it says on startup.
```

Comment:
I am a little lost here. Are these accurate statements? 
1. The baseline node configuration YAML does not contain any information about the node being validated. It only contains the information (such as node address, metrics_port, etc.) of the baseline node (the "golden" node against which the user's node is being validated). In this present context, we prefer that these baseline nodes are Aptos nodes, i.e., we use our own Aptos nodes to measure/validate how good the user node is. 
2. If yes, then I didn't get "modify certain fields, such as the node address or evaluator set used." Why would a user need to modify the node address? Is it because the user may want to use a non-Aptos node as a baseline node? 
3. I think I am missing the intuition for what a baseline node configuration is? Or is (1) above is correct?

Reply:
1. This is correct, the baseline configuration only describes where to find the baseline, what evaluations to run, and configuration for those evaluations. It configures nothing about the node under investigation, as that will likely change with each request.
2. Realistically for some configurations, e.g. the devnet fullnode case, our baseline isn't great because we don't have our metrics port open. That's what I'm talking about. Another example is the mint key for the TPS test case, that could warrant a whole section of its own. It'd be great to actually have a separate document that enumerates the important configuration fields and how to set them, but that's out of scope for this README imo.
3. You understanding in 1 is correct.